File: aspnet-core/src/Player.Application.Contracts/Restaurants/RestaurantDto.cs
using Player.Items;
using System;
using System.Collections.Generic;
using System.Text;

namespace Player.Restaurants
{
    public class RestaurantDto
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int Star { get; set; }
        public string Open { get; set; }
        public string Close { get; set; }
        public string ImageUrl { get; set; }
        public string WebUrl { get; set; }
        public List<Item> Items { get; set; }
        public RestaurantDto(string id, string name, int star, string open, string close, string imageUrl, string webUrl)
        {
            Id = id;
            Name = name;
            Star = star;
            Open = open;
            Close = close;
            ImageUrl = imageUrl;
            WebUrl = webUrl;
            Items = new List<Item>();
        }
        public RestaurantDto() { }
    }
}

File: angular/src/app/modules/orders/models/restaurant.dto.ts
export interface RestaurantDTO {
    id: string
    name: string
    star: number
    open: string
    close: string
    imageUrl: string
    webUrl: string
    items: ItemDTO[]
  }
  
  export interface ItemDTO {
    id: string
    name: string
    quantity: number
    imageUrl: string
    priceOrigin: number
    priceDiscount: number
    optionGroups: OptionGroupDTO[]
  }
  
  export interface OptionGroupDTO {
    id: string
    name: string
    isAvailable: boolean
    selectMin: number
    selectMax: number
    options: OptionDTO[]
  }
  
  export interface OptionDTO {
    id: string
    name: string
    isAvailable: boolean
    price: number
  }

  export interface ItemCardDTO {
    id: string
    name: string
    quantity: number
    total: number
  }
  export interface CardDTO {
    quantity: number
    item: ItemCardDTO
  }


  export interface UserOrderDTO {
    restaurantId: string,
    groupOrderId: string,
    itemCountAndIds: ItemDTO[]
    optionCountAndIds: ItemOptionDTO[]
    note: string
}

export interface ItemOrderDTO {
    id: string,
    count: number
}

export interface ItemOptionDTO {
    id: string,
    count: number
}

  